pyartcd/pyartcd/pipelines/gen_assembly.py

```
@@ -0,0 +1,99 @@
+
+        cmd = cmd.replace(" *\\\n *", " ").replace(" *\n *", " ").strip()  # Cleaning whitespace
+
+        # Run doozer command
+        output_all = exectools.cmd_gather(f"doozer --assembly=stream {cmd}")
+        if output_all[0]:
+            print(f"Error: {output_all[0]}")
+            print(output_all[2])
+            sys.exit(1)
+
+        output = output_all[1]
+        release_data = output[output.find("releases:"):].strip()
+        print(release_data)
+
+        base = f"openshift-{self.build_version}"  # The base that we have to branch off of
+        temp_branch = f"automation_{self.build_version}_{self.build_number}"  # format of the temporary branch
+        github_client = github.GithubClient(base=base, data=release_data, assembly_name=self.assembly_name,
+                                            temp_branch=temp_branch)
+
+        try:
+            # Use the output to create a PR
+            if github_client.branch_exists():  # If a branch exists with the same name as the temp branch
+                raise exceptions.BranchExists(
+                    "Temporary branch already exists. Cannot push changes. Delete the temp branch and try again")
+            github_client.sync_with_upstream()  # Sync base branch with upstream
+            github_client.create_branch()  # Create the temporary branch off of the base branch
+            content = github_client.update_file()  # add and update changes
+            github_client.push_change(content)
+            pr_url = github_client.create_pr()
+
+            # Send message to the corresponding release channel
+            slack_client = self.runtime.new_slack_client()
+            slack_client.bind_channel(f"{self.build_version}")
+            await slack_client.say(
+                f"TEST: Gen-assembly job for *{self.build_version}* successful. PR created: {pr_url}")
+        except (exceptions.GithubApiExceptions, Exception) as e:
+            print(f"Error: {e}")
+            sys.exit(1)
```

Print out the error message is not actual exception handling. It will also hide the stacktrace and original exception message. Please remove this try-except block and let it crash when there's an exception.

---

Understood! Done!